AWSIoTTests/AWSIoTDataTests.m

```diff
@@ -995,17 +995,31 @@ - (void)testUsernameMetaData {
     
     // Check state after adding additional fields
     NSDictionary<NSString *,NSString *> * metaData = @{@"foo": @"bar", @"clazz": @"2"};
-    [iotDataManager addUserMetaData: metaData];
-    NSString *expectedUserMetaData = [NSString stringWithFormat:@"?SDK=iOS&Version=%@&foo=bar&clazz=2", AWSIoTSDKVersion];
+    [iotDataManager updateUserMetaData: metaData];
+    NSString *defaultUserMetaData = [NSString stringWithFormat:@"?SDK=iOS&Version=%@", AWSIoTSDKVersion];
     NSString *actualUserMetaData = [[iotDataManager mqttClient] userMetaData];
-    XCTAssertTrue([actualUserMetaData isEqualToString:expectedUserMetaData]);
-    
+    XCTAssertTrue([actualUserMetaData hasPrefix:defaultUserMetaData] && [actualUserMetaData containsString:@"foo=bar"] && [actualUserMetaData containsString:@"clazz=2"]);
```

Comment:
Break each of these conditions into its own `XCTAssert` rather than combining them, so it's easier to tell which one fails.

Reply:
Done